src/app/shared/interceptors/http-error-interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+  HTTP_INTERCEPTORS,
+} from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import { catchError, EMPTY, Observable, retry } from 'rxjs';
+
+import { API_URL } from '@core/env.token';
+import { ToastFacadeService } from '@shared/services';
+
+@Injectable()
+class HttpErrorInterceptor implements HttpInterceptor {
+  private base_url = inject(API_URL);
+  private toastService = inject(ToastFacadeService);
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    const clone = request.clone({
+      url: `${this.base_url}${request.url}`,
+    });
+
+    return next.handle(clone).pipe(
+      retry(1),
+      catchError((error: HttpErrorResponse) => {
+        let errorMsg = '';
+
+        if (error.error instanceof ErrorEvent) {
+          errorMsg = `Error: ${error.error.message}`;
+        } else {
+          errorMsg = `Error Code: ${error.status},  Message: ${error.message}`;
+        }
+
+        if (error.status >= 500) {
+          this.toastService.showError('Błąd serwera', 'Błąd');
```

Comment:
fajny ten bład, taki nic nie mówiący xd

a tak serio musimy to zhandlować jakoś. Albo dostajemy kody błędu z backendu i wtedy mamy słownik na froncie i w zależności od błędu wyświetrlamy stosowny komunikat lub robimy konkretną akcję. Tak czy siak, trzeba to przegadać prośba o wrzucenie ticketu do backlogu, oznaczyć go że wymaga doprecyzowania, wrzucić komentarz z TODO i linkiem do taska

a tak to klasa interceptor 💪